### PR TITLE
For avatar picker restrict only with image file types

### PR DESCRIPTION
**Pull request by Santhosh-Sellavel**
@tgolen 

### Details
Restricted Avatar picker to pick only image file types.

### Fixed Issues

$ #4645 

### Tests && QA Steps
1. Go to Settings
2. Click Profile
3. Click the pen icon to update or add a photo
4. Click upload photo
5. Refer **Fix** section below, it should work as shown there.
6. Create a new workspace if you don't have one.
7. Go to Setting 
8. Click on your workspace.
9. Tap on the Workspace icon  
10. Click the pen icon to update or add a photo
11. Click upload photo
12. Refer **Fix** section below, it should work as shown there.

##### In Desktop or web:
The file picker opens and will allow picking only image file types now.

##### In Mobile, we should show only two options now
- Take photos,
- Gallery,
-  ~Choose Document~ Should be removed 

#### Mobile web
Options may vary based on the OS Type - but should be able to pick images only
Select Gallery should show only the photos

### Tested On
- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screenshot 2021-09-05 at 4 35 34 PM
Screenshot 2021-09-05 at 4 34 06 PM

#### Mobile Web
Simulator Screen Shot - iPhone 11 - 2021-09-05 at 18 11 02
Simulator Screen Shot - iPhone 11 - 2021-09-05 at 18 11 06

#### Desktop
Screenshot 2021-09-05 at 4 21 48 PM
Screenshot 2021-09-05 at 4 21 03 PM

#### iOS
Simulator Screen Shot - iPhone 12 - 2021-09-05 at 16 31 18
Simulator Screen Shot - iPhone 12 - 2021-09-05 at 16 30 02

#### Android
Screenshot_1630839652
Screenshot_1630839665

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@tgolen Waiting for your initial review if you are okay, I'll mark it as Ready for review.

File: src/components/AttachmentPicker/index.native.js
Comment (tgolen) [inline src/components/AttachmentPicker/index.native.js:99]: This is probably worth adding a comment to explain why this logic is being done

**Review by tgolen — CHANGES_REQUESTED** [review]
Looks good! This is mostly just small little stuff that I noticed.

File: src/components/AttachmentPicker/index.native.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.native.js:99]: Sure!

File: src/components/AttachmentPicker/index.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js:8]: Is this right `@returns {String|undefined}` ?

Can you help me here if it's not right,  what to add here?

File: src/components/AttachmentPicker/index.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js:16]: Will do

File: src/components/AttachmentPicker/AttachmentPickerPropTypes.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/AttachmentPickerPropTypes.js:24]: Ya sure!

File: src/components/AttachmentPicker/index.js
Comment (tgolen) [reply src/components/AttachmentPicker/index.js:8]: Yep, I think that's it! In PHP we use `?String` but I was looking and I don't think that is a thing in JSDoc.

File: src/components/AttachmentPicker/AttachmentPickerPropTypes.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/AttachmentPickerPropTypes.js:24]: /** Type associated with the Picker's file type(s) */ 
Is this good enough?

File: src/components/AttachmentPicker/index.native.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.native.js:99]: `// Excluding choose document option for picker type - image`

File: src/components/AttachmentPicker/AttachmentPickerPropTypes.js
Comment (tgolen) [reply src/components/AttachmentPicker/AttachmentPickerPropTypes.js:24]: That's pretty good, yeah. If I was writing it, I think I would say:

```
/** The types of files that can be selected with this picker. */
```

File: src/components/AttachmentPicker/index.native.js
Comment (tgolen) [reply src/components/AttachmentPicker/index.native.js:99]: The best strategy for writing a comment is to describe the "why" and this comment is only describing "what". A better comment could be:

`// When selecting an image on a native device, it would be redundant to have a second option for choosing a document, so it is excluded in this case.`

File: src/components/AttachmentPicker/AttachmentPickerPropTypes.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/AttachmentPickerPropTypes.js:24]: > /** The types of files that can be selected with this picker. */

This is more meaningful!

File: src/components/AttachmentPicker/index.native.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.native.js:99]: Thanks for your guidance!

File: src/components/AttachmentPicker/index.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js:16]: @tgolen 
After removing the undefined following error shown, 
Screenshot 2021-09-07 at 11 21 12 PM

Screenshot 2021-09-07 at 11 20 51 PM

I think it's better to leave `undefined` as it is, what to do now?

Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js:16]: Or remove the return statement?

Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js]: Removing the `return` statement implies undefined so removed it.

Comment (tgolen) [reply src/components/AttachmentPicker/index.js]: Ah yeah, I think removing the `return` entirely is fine 👍

File: src/components/AttachmentPicker/index.js
Comment (tgolen) [inline src/components/AttachmentPicker/index.js:15]: NAB (not a blocker), but this comment could probably be moved up to the JSDoc for `@returns`

**Review by tgolen — DISMISSED** [review]

File: src/components/AttachmentPicker/index.js
Comment (Santhosh-Sellavel) [reply src/components/AttachmentPicker/index.js]: Updated

**Review by tgolen — DISMISSED** [review]
@marcaaron it's all yours

**Comment by Santhosh-Sellavel** [comment]
Resolved a conflict.

**Review by tgolen — APPROVED** [review]

**Review by marcaaron — APPROVED** [review]
This LGTM I had a small comment about the comment left in the native.js file, but I don't think it's a blocker.

File: src/components/AttachmentPicker/index.native.js
Comment (tgolen) [reply src/components/AttachmentPicker/index.native.js:101]: Ah, great point, Marc!

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-20.  :confetti_ball:

**Comment by Jag96** [comment]
I'm seeing this warning on `main` when running the app on iOS and it looks like it's related to this PR, @Santhosh-Sellavel can you have a look?

image

**Comment by Santhosh-Sellavel** [comment]
@Jag96 I'll check again to ensure, but I saw this before this PR is merged.

**Comment by Santhosh-Sellavel** [comment]
> I'm seeing this warning on `main` when running the app on iOS and it looks like it's related to this PR, @Santhosh-Sellavel can you have a look?

@Jag96 this is caused by this PR only.  I don't know how missed it.
Also wondering how @marcaaron 👀 missed it.

Need to change the import line 
https://github.com/Expensify/App/blob/3daf85f6733a828c6d6e8b3b3402245206c19c1c/src/components/AttachmentPicker/index.native.js#L8

as below 
```js 
import {propTypes as basePropTypes, defaultProps} from './AttachmentPickerPropTypes';
```

And need to set `defaultProps` alone the following lines.
https://github.com/Expensify/App/blob/3daf85f6733a828c6d6e8b3b3402245206c19c1c/src/components/AttachmentPicker/index.native.js#L304-L305

As below.
```js
AttachmentPicker.defaultProps = defaultProps;
```
Can I start the PR?

**Comment by Jag96** [comment]
@Santhosh-Sellavel yes feel free to start on a PR, just make sure you add regression testing to confirm the tests from this PR still pass

**Pull request merged**
